File: source/lang/dart/rabbit_template.dart
String uni2zg(String output) {
  List<Map<String, String>> rules = {{UNI2ZG}};
  return replace_with_rule(rules, output);
}

String zg2uni(String output) {
  List<Map<String, String>> rules = {{ZG2UNI}};
  return replace_with_rule(rules, output);
}

String replace_with_rule(List<Map<String, String>> rules, String input) {
  rules.forEach((rule) {
    input = input.replaceAllMapped(RegExp(rule["from"]), (match1) {
      return rule["to"].replaceAllMapped(RegExp(r"#(\d{1})"), (match2) {
        var t = match1.group(int.parse(match2.group(1)));
        return t == null ? "" : t;
      });
    });
  });
  return input;
}

File: source/lang/typescript/rabbit_template.php
<?php

/**
 * Rabbit is Another Zawgyi <=> Unicode Converter.
 * 
 * @author Saturngod <me@saturngod.net>
 */
class Rabbit
{
    /**
     * Rabbit Converter version.
     *
     * @var string
     */
    const VERSION = '{{VERSION}}';
    
    /**
     * Convert unicode string to zawgyi.
     *
     * @param  string $unicode
     * @return string
     */
    public static function uni2zg($unicode)
    {
        $rule = json_decode("{{UNI2ZG}}", true);

        return self::replaceWithRule($rule, $unicode);
    }

    /**
     * Convert zawgyi string to unicode.
     *
     * @param  string $unicode
     * @return string
     */
    public static function zg2uni($zawgyi)
    {
        $rule = json_decode("{{ZG2UNI}}", true);

        return self::replaceWithRule($rule, $zawgyi);
    }

    /**
     * Replace the line break to character to parse the data correctly
     * @param string $string
     * @return string
     */
    private static function parseline($string){
        $string = str_replace(chr(10), "\\n", $string);
        $string = str_replace(chr(13), "\\n", $string);
        $string = str_replace("\f", "\\f", $string);
        return $string;
    }

    /**
     * Replace the string with rules.
     *
     * @param  array $rule
     * @param  string $output
     * @return string
     */
    protected static function replaceWithRule($rule, $output)
    {
        foreach ($rule as $data) {

            $from_json = $data["from"];

            //search line break.
            //if line break include , need to fix the line
            if (strpos($from_json, chr(13)) !== false) {
                $from_json = self::parseline($from_json);
            }

            $from = "~".json_decode('"'.$from_json.'"')."~u";
            $to = json_decode('"'.$data["to"].'"');
            $output = preg_replace($from, $to, $output);
        }

        return $output;
    }
}
